tools/extract_pdfs.py

```
@@ -0,0 +1,204 @@
+            text = page.get_text()
+            
+            if text.strip():  # Only add pages with content
+                pages_content.append({
+                    'page_number': page_num + 1,
+                    'text': clean_text(text)
+                })
+        
+        doc.close()
+        return pages_content
+        
+    except Exception as e:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)